core/client/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -319,7 +319,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
       // Canceled is considered a valid status and handled in the writer. We avoid creating a
       // CanceledException as an optimization.
       if (response.getMessage().asResponse().getStatus() != PStatus.CANCELED) {
-        response.unwrapException();
+        CommonUtils.unwrapResponse(response.getMessage().asResponse());
```

Comment:
I've found a way to work around it:
Whenever the server returns error, use type Response. Otherwise, return the actual data (that has no status).